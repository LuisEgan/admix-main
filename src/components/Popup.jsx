import React from "react";
import PropTypes from "prop-types";

class Popup extends React.PureComponent {
  closePopup() {
    const { togglePopup } = this.props;

    if (!togglePopup && process.env.NODE_ENV === "development") {
      console.error("Warning! No togglePopup() function was provided to Popup");
    } else {
      togglePopup();
    }
  }

  render() {
    const { children, showPopup, id } = this.props;

    if (!showPopup) {
      return null;
    }

    return (
      <div id={id} className="popup">
        <div className="popup-inner">{children}</div>
        <div className="popup-bg" onClick={this.closePopup.bind(this)} />
      </div>
    );
  }
}

Popup.propTypes = {
  showPopup: PropTypes.bool.isRequired,
  togglePopup: PropTypes.func
};

Popup.defaultProps = {
  showPopup: false
};

export default Popup;
