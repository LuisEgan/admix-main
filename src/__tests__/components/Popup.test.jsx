import React from "react";
import { shallow, mount } from "enzyme";
import Popup from "../../components/Popup";

describe("<Popup />", () => {
  it("does not render Popup initially", () => {
    const wrapper = shallow(<Popup />);
    expect(wrapper.find(".popup").exists()).toBe(false);
  });

  it("does render only when showPopup prop is true", () => {
    const wrapper = shallow(<Popup showPopup={false} />);
    expect(wrapper.get(0)).toBeNull();

    wrapper.setProps({ showPopup: true });
    expect(wrapper.find(".popup").length).toBe(1);
  });

  it("matches snapshot", () => {
    const wrapper = shallow(<Popup showPopup={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("runs togglePopup() when the background is clicked if togglePopup prop exists", () => {
    const mockTogglePopup = jest.fn();

    const wrapper = shallow(
      <Popup showPopup={true} togglePopup={mockTogglePopup} />
    );
    const bg = wrapper.find(".popup-bg");
    bg.simulate("click");

    expect(mockTogglePopup.mock.calls.length).toEqual(1);
  });
});
