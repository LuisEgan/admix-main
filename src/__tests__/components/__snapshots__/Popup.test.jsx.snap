// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup /> matches snapshot 1`] = `
<div
  className="popup"
>
  <div
    className="popup-inner"
  />
  <div
    className="popup-bg"
    onClick={[Function]}
  />
</div>
`;
